File: admin/lib/core/models/persona_fisica.dart
import 'id_document_type.dart';

/// Datos de identificación de una persona física relacionada con un
/// Cliente (Apoderado Legal o Beneficiario Controlador) — no confundir
/// con [Cardholder], que es una persona física que usa una Tarjeta.
/// Comparten el mismo shape a propósito para no duplicar la definición
/// de campos tres veces — ver docs/business/kyb-cliente.md, "Diseño:
/// reutilizar un shape común de persona física".
class PersonaFisica {
  final String fullName;
  final IdDocumentType idDocumentType;
  final String idDocumentNumber;
  final String? curp;
  final String? rfc;

  const PersonaFisica({
    required this.fullName,
    this.idDocumentType = IdDocumentType.ine,
    required this.idDocumentNumber,
    this.curp,
    this.rfc,
  });

  PersonaFisica copyWith({
    String? fullName,
    IdDocumentType? idDocumentType,
    String? idDocumentNumber,
    String? curp,
    String? rfc,
  }) {
    return PersonaFisica(
      fullName: fullName ?? this.fullName,
      idDocumentType: idDocumentType ?? this.idDocumentType,
      idDocumentNumber: idDocumentNumber ?? this.idDocumentNumber,
      curp: curp ?? this.curp,
      rfc: rfc ?? this.rfc,
    );
  }
}

File: backend/internal/domain/shared/persona_fisica.go
package shared

// IDDocumentType — compartido entre el KYC de un Cardholder
// (internal/domain/cardholder) y el KYB de un Cliente
// (internal/domain/client, PersonaFisica de un Apoderado/Beneficiario) —
// mismo shape a propósito, ver
// admin/lib/core/models/persona_fisica.dart, "Diseño: reutilizar un
// shape común de persona física".
type IDDocumentType string

const (
	IDDocumentTypeINE               IDDocumentType = "INE"
	IDDocumentTypePasaporte         IDDocumentType = "pasaporte"
	IDDocumentTypeCedulaProfesional IDDocumentType = "cedula_profesional"
)

// PersonaFisica — datos de identificación de un Apoderado Legal o
// Beneficiario Controlador de un Cliente. No confundir con Cardholder
// (una persona física que usa una Tarjeta) — ver
// admin/lib/core/models/persona_fisica.dart.
type PersonaFisica struct {
	FullName         string
	IDDocumentType   IDDocumentType
	IDDocumentNumber string
	CURP             *string
	RFC              *string
}
